### ISSUE TITLE
Registering a factory for only a subset of conditions

### ISSUE THREAD

**Issue by Mr0grog** [rgeo/rgeo-activerecord#33]
I am working with a client whose system needs to use the `uses_lenient_assertions` option when constructing RGeo objects. (They've got some bad data in the DB and cleaning it out is a whole other project, which obviously does need to happen at some point, but isn't happening now.)

To handle that, I thought I'd try and target just objects with the SRID 4326:

``` rb
RGeo::ActiveRecord::SpatialFactoryStore.instance.tap do |config|
  config.register(
    RGeo::Geographic.spherical_factory(srid: 4326, uses_lenient_assertions: true),
    srid: 4326)
end
```

But it turns out that this actually registers the factory with all the default conditions, except the `srid` condition is set to `4326`. So this didn't match because the column is a `geography` and not a `geometry` (the default value for `sql_type` in `config.register`).

It'd be nice if there were a way to instead have the above code cause this factory to be used for _any_ column with an SRID of 4326 (maybe with lower priority than a factory registered with > 1 condition or something).

At the very least, though, I would have appreciated it if the docs were a little clearer on what `config.register` actually does. I had to do some digging in the source to understand why the code above wasn't working.

**Comment by keeperhood** [comment]
I have similar issues with a factory registered with `config.register` never gets used automagically. 

In my `config/initializers/postgis.rb` I have
```ruby
RGeo::ActiveRecord::SpatialFactoryStore.instance.tap do |config|
  config.default = RGeo::Geographic.spherical_factory(srid: 4326)
  config.register(
    RGeo::Geographic.simple_mercator_factory.projection_factory,
    sql_type: 'geometry',
    geo_type: 'polygon',
    srid: 3875
  )
end
```

On column types that are `geometry(Polygon,3857)` Postgis type i want to use the 2nd (non-default) factory. While on all the other spatial column types which are `geography(Polygon,4326)` I want to use the default factory.

With the config above always the spherical_factory with the 4326 from the `config.default` is used always for both of the mentioned postgis column types. The simple_mercator never seems to be recognised by the `geometry(Polygon, 3857)` RGeo activerecord spatial columns when loading data from the db.

@Mr0grog What did you find out after digging in the source code? Do you understand what `config.register` actually does?

**Comment by keeperhood** [comment]
I have a Layer model which has a bbox column which is of type geometry(Polygon,3857) in the db. So I've tried to inspect the column definitions of the Layerclass like this 

    Layer.columns

And  my bbox column had contained this definition structure:

    => #<ActiveRecord::ConnectionAdapters::PostGIS::SpatialColumn:0x007fc21bb32d08
     @array=false,
     @cast_type=
      #<ActiveRecord::ConnectionAdapters::PostGIS::OID::Spatial:0x007fc21bb388c0
       @geo_type="Polygon",
       @has_m=false,
       @has_z=false,
       @spatial_factory=#<Proc:0x007fc2197df7c0@/Users/igor/work/ac/agricircle-giscockpit-api/config/initializers/postgis.rb:3>,
       @sql_type="geometry(Polygon,3857)",
       @srid=3857>,
     @default=nil,
     @default_function=nil,
     @geographic=false,
     @geometric_type=RGeo::Feature::Polygon,
     @has_m=false,
     @has_z=false,
     @limit={:srid=>3857, :type=>"polygon"},
     @name="bbox",
     @null=false,
     @sql_type="geometry(Polygon,3857)",
     @srid=3857,
     @table_name="layers">

Then I've tried to copy the @geo_type @sql_type and @srid strings into the config/initializers/postgis.rb

    RGeo::ActiveRecord::SpatialFactoryStore.instance.tap do |config|
      config.default = RGeo::Geographic.spherical_factory(srid: 4326)
    
      config.register(
        RGeo::Geographic.simple_mercator_factory.projection_factory,
        sql_type: 'geometry(Polygon,3857)',
        geo_type: 'Polygon',
        srid: 3857
      )
    end

This seems to have solved the issue, perhaps the strings need to be exact and I've noticed that I've mistyped 3875 instead of 3857.

**Comment by Mr0grog** [comment]
@keeperhood oof, it’s been quite a while since I had to address this.

IIRC, yeah, the strings have to match *exactly* how it’s defined in the DB, so depending how you created a column, `sql_type` might have to be `geometry(Polygon,3857)` or `geometry(Polygon)` or `geometry(Polygon,4326)` or whatever.

Looking back at the old source, it looks like my solution was basically the same as yours; just register with extremely specific conditions.

**Comment by keithdoggett** [comment]
Fixed by #62
